docker-compose.yml

```diff
@@ -457,8 +457,8 @@ services:
     volumes:
       - type: bind
         read_only: true
-        source: ./nginx
-        target: /etc/nginx
+        source: ./nginx/nginx.conf
```

Comment:
Then `/etc/nginx/mime.types` wouldn't exists on that directory. Do you have any suggestion for this?

Reply:
No :)

> `docker-compose` makes some problems bind mounting a single file, I'd suggest not changing this.

It's OK here though as I think more about it. No other process is going to use this file and no other process is going to change this file anyway.

Reply:
Shall we mount the directory to some other path and do something like this instead then: https://stackoverflow.com/a/30152496/90297

As it would be quite confusing for someone to realize `nginx` is the odd one out where we only mount a single file in case we want to add something new in the future?

Reply:
That would be a worse hack than this IMHO.

Reply:
@aminvakil why do you think so?

Reply:
AFAIK the main reason, mounting a single file is bad in docker, is when you want to change that file on the host and see your changes in container, yeah, that would not happen.

But I think here it's ok as nginx container should be restarted if the file (`nginx.conf` here) has been changed in host anyway, so it's fine IMO.

Reply:
That's not my concern at all. For every other service where we have a config directory in this repo, we mount the entire directory and expect anything there to be available in the container. A change like this would break that convention.

Reply:
Ok, what about making this an exception to increase its readibility?

Reply:
Could you please state the other option clearer, I think I have not understood your solution correctly, use another nginx docker image?

Reply:
I'm just proposing to keep the directory mounting behavior but instead of mounting it directly to `/etc/nginx` mount it somewhere else such as `/opt/nginx-conf` or something and then use the `CMD` or entry point override to point Nginx to use config from there.

Reply:
I'm against what @BYK's proposing, I think just mounting the file directly would be simpler and much less problematic (see some issues that people can't build their own Docker image since they're behind some sort of hard network proxy/firewall).

Reply:
@BYK gotcha, thanks for clarification.

Both ways looks good to me.

Reply:
>see some issues that people can't build their own Docker image since they're behind some sort of hard network proxy/firewall

I wasn't recommending us to build our own image, you can override `CMD` and `ENTRYPOINT` from `docker-compose.yml`.

Anyway, may I offer a compromise then:

1. Move the `nginx.conf` file outside of the `nginx` directory
2. Mount that file by itself

It achieves what we want both: hassle-free and not confusing for any onlookers or future maintainers.

WDYT @aldy505?

Reply:
My proposal is to move to the top level instead of a dedicated directory to avoid confusion around mounting.

Reply:
This is the best way to go IMHO as well.

Reply:
Alright, will do that.